### PR TITLE
MethodGenerator::fromArray() add config keys

### PR DESCRIPTION
**Pull request by Brammm**
Continuation of https://github.com/laminas/laminas-code/pull/107

### CODE REVIEW

File: test/Generator/MethodGeneratorTest.php [hunk 1]
```diff
@@ -307,6 +307,37 @@ public function testCreateFromArray()
...
+        $methodGenerator = MethodGenerator::fromArray([
+            'name'             => 'SampleMethod',
+            'returnsreference' => $value,
+        ]);
+
+        self::assertSame($expected, $methodGenerator->isReturnsReference());
+    }
+
+    /**
+     * @return string[][]
```
Comment (Brammm) [inline test/Generator/MethodGeneratorTest.php:326]: Just realized this type is wrong (copied from somewhere else and then forgotten about), though not sure about conventions in the project as to what I should change it to.

File: src/Generator/MethodGenerator.php [hunk 2]
```diff
@@ -310,6 +314,11 @@ public function setReturnsReference($returnsReference)
         return $this;
     }
 
+    public function isReturnsReference(): bool
```
Comment (Ocramius) [inline src/Generator/MethodGenerator.php:317]: The `is` should really be removed here - naming is totally wonky (will rename on merge)

File: test/Generator/MethodGeneratorTest.php (hunk 1 above)
Comment (Ocramius) [reply test/Generator/MethodGeneratorTest.php:326]: I'll adjust this on merge :+1:

**Review by Ocramius — APPROVED** [review]

**Comment by Ocramius** [comment]
Merged via #121, thanks @Brammm!

**Pull request closed without merging**